### PR TITLE
[PROJECT] Outpost of Cogs

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Adds new functionality when a blood cultist is drained at a vitality sigil.

When the cultist dies from the vitality sigil, a special message will be broadcasted to all clock cultists. Shortly after, a portal will open up where the cultist died. Only clock cultists are able to enter the portal.

image

Going through the portal leads you to the "Outpost of Cogs", a far cry from the holy city of Reebe itself. Five minutes after the portal has been open, a centcomm announcement will play as the portal shifts colors, allowing anyone to enter.

To leave the Outpost, there's a portal opposite where you entered, on the north side.

image

image

Original map by dawson

## How This Contributes To The Skyrat Roleplay Experience
Adds depth to the clock cult vs blood cult interaction

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
See above

</details>

### CODE REVIEW

File: modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+
+/// Spawn reebe with the appropriate fanfare after sacrificing a blood cultist
+/proc/spawn_reebe(atom/target_atom)
+	set waitfor = FALSE
+
+	if(GLOB.reebe_began_spawning)
+		return
+
+	GLOB.reebe_began_spawning = TRUE
```
Comment (MrMelbert) [inline modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm]: Should really just be a static var if you don't use the global scope anywhere else

File: modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+	var/turf/atom_turf = get_turf(target_atom) // Protection in case this gets deleted
+	var/area/atom_area = get_area(target_atom)
+
+	sleep(2 SECONDS)
```
Comment (MrMelbert) [inline modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm:21]: Sure are a lot of sleeps without sanity checks here...

File: modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm (hunk 1 above)
Comment (Zonespace27) [reply modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm]: Done

File: modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm (hunk 2 above)
Comment (Zonespace27) [reply modular_skyrat/modules/clock_cult/code/outpost_of_cogs.dm:21]: Fixed

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
Quick things to note.

<details>
<summary> Map review </summary>

I will strangle you. We HAVE bronze tables in code.
Don't like them? Honestly, not my concern or problem.

image

I also never really played much with Clockies, but the ruin? Area? Whatever, could use more environmental fluff. It feels empty.

image


<details>

**Review by Jolly-66 — APPROVED** [review]
Map looks better.

Although, could use dirt cleanables for the more "old" vibe but I won't hold this PR for not having that.

**Comment by Zonespace27** [comment]
Dirtied up the place a bit, added some cult runes

File: modular_skyrat/modules/clock_cult/code/turf.dm [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/turf/open/indestructible/reebe_void
+	name = "void"
+	desc = "A white, emptyw void, quite unlike anything you've seen before."
```
Comment (Paxilmaniac) [inline modular_skyrat/modules/clock_cult/code/turf.dm]: emptyw

Comment (Zonespace27) [inline modular_skyrat/modules/clock_cult/code/turf.dm]:
```suggestion
	desc = "A white, empty void, quite unlike anything you've seen before."
```

Comment (Zonespace27) [reply modular_skyrat/modules/clock_cult/code/turf.dm]: Emptyw (fixed)

**Review by Paxilmaniac — APPROVED** [review]
looks good to me

**Pull request merged**